app/models/portfolio_item.rb

```diff
@@ -24,4 +27,11 @@ def resolved_workflow_refs
   def item_workflow_ref
     [self, portfolio].detect(&:workflow_ref)&.workflow_ref
   end
+
+  def validate_workflow
+    Approval.call_workflow_api { |api| api.show_workflow(workflow_ref) }
+  rescue Catalog::ApprovalError
```

Comment:
I'm not sure I think we should use `rescue` to modify behavior.  It would probably be best to have a boolean Approval endpoint validating the workflow. 
```
def validate_workflow
   unless Approval.call_workflow_api { |api| api.workflow_valid(workflow_ref) }
     errors.add(:workflow, "invalid")
   end
  true
end
```
 @bzwei what do you think of this suggestion?

Reply:
I am not sure whether this validation is needed at all. Basically you want to verify whether a referenced entity still exists in another service. Do all APIs need to provide an "are you still there" endpoint for every resource? The answer is likely not. 
It is not very useful to validate here because it can still fail at a later time if the resource is removed after the validation. I would say let it fail naturally at runtime.

Reply:
I'm ok with not validating `workflow_ref` due to the fact that it could fail and we could just handle it at runtime. But we have had bugs assigned due to the API not validating `workflow_ref` on update, are these bugs invalid then? 

ex: https://projects.engineering.redhat.com/browse/SSP-560

Reply:
@gmcculloug I know we had discussed talking more about this concept of internal microservice validation.  Per @bzwei suggestion he is fine with it failing naturally at runtime.  Does any one else ( @mkanoor ) have thoughts on this? I'd like to get this PR moving again.

Reply:
@syncrou Is this about converting refs to strings or an edge case where the API can pass in random values and we expect it to be 422 error code

Reply:
@mkanoor the latter. The discussion is: should catalog be rescuing everything and re throwing our own 'correct' error? Or Should the Approval service ( or any other service ) provide a method that exposes their own boolean validation for specific endpoints?

Though, we can discuss on Thursday if the meeting ends up working out. :)

Reply:
@lindgrenj6 
When we do a post we have to make sure that the data is sent as a JSON payload as opposed to a string. If it is sent as a string it will loose the datatype

The post has an option called :as => :json which converts the params to JSON and also sets the CONTENT_TYPE to application/json